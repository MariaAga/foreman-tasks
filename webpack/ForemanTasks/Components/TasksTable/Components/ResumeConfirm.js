import React from 'react';
import PropTypes from 'prop-types';
import { sprintf, translate as __ } from 'foremanReact/common/I18n';
import { Button } from 'patternfly-react';
import ForemanModal from 'foremanReact/components/ForemanModal';
import {
  CANCEL_CONFIRM_MODAL_ID,
  RESUME_CONFIRM_MODAL_ID,
  CANCEL_SELECTED_CONFIRM_MODAL_ID,
  RESUME_SELECTED_CONFIRM_MODAL_ID,
} from '../TasksTableConstants';

export const ResumeConfirm = ({ closeModal, action, selectedRowsLen, id }) => (
  <ForemanModal title={__('Resume Selected Tasks')} id={id}>
    {sprintf(
      __(
        `This will resume %s task(s), putting them in the running state. Are you sure?`
      ),
      selectedRowsLen
    )}
    <ForemanModal.Footer>
      <Button onClick={closeModal}>{__('No')}</Button>
      <Button
        bsStyle="primary"
        onClick={() => {
          action();
          closeModal();
        }}
      >
        {__('Yes')}
      </Button>
    </ForemanModal.Footer>
  </ForemanModal>
);

ResumeConfirm.propTypes = {
  closeModal: PropTypes.func.isRequired,
  selectedRowsLen: PropTypes.number.isRequired,
  action: PropTypes.func,
  id: PropTypes.oneOf([
    CANCEL_CONFIRM_MODAL_ID,
    RESUME_CONFIRM_MODAL_ID,
    CANCEL_SELECTED_CONFIRM_MODAL_ID,
    RESUME_SELECTED_CONFIRM_MODAL_ID,
  ]).isRequired,
};

ResumeConfirm.defaultProps = {
  action: () => null,
};

export default ResumeConfirm;
