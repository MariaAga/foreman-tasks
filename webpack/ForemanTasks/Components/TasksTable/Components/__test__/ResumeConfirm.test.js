import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { ResumeConfirm } from '../ResumeConfirm';
import {
  RESUME_CONFIRM_MODAL_ID,
  RESUME_SELECTED_CONFIRM_MODAL_ID,
} from '../../TasksTableConstants';

const baseProps = {
  closeModal: () => null,
  selectedRowsLen: 3,
  action: () => null,
};
const fixtures = {
  'renders RESUME modal': {
    ...baseProps,
    id: RESUME_CONFIRM_MODAL_ID,
  },
  'renders RESUME_SELECTED modal': {
    ...baseProps,
    id: RESUME_SELECTED_CONFIRM_MODAL_ID,
  },
};

describe('ResumeConfirm', () =>
  testComponentSnapshotsWithFixtures(ResumeConfirm, fixtures));
