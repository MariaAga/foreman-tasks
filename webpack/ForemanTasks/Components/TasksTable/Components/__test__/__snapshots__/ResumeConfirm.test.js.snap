// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeConfirm renders RESUME modal 1`] = `
<ForemanModal
  id="resumeConfirmModal"
  title="Resume Selected Tasks"
>
  This will resume 3 task(s), putting them in the running state. Are you sure?
  <Component>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      No
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      Yes
    </Button>
  </Component>
</ForemanModal>
`;

exports[`ResumeConfirm renders RESUME_SELECTED modal 1`] = `
<ForemanModal
  id="resumeSelectedConfirmModal"
  title="Resume Selected Tasks"
>
  This will resume 3 task(s), putting them in the running state. Are you sure?
  <Component>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
    >
      No
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      Yes
    </Button>
  </Component>
</ForemanModal>
`;
